From ca8a39d1bb999ba98ead45e53041fc3ea28ecd5c Mon Sep 17 00:00:00 2001
From: David Woodhouse <David.Woodhouse@intel.com>
Date: Thu, 10 Nov 2011 23:19:32 +0000
Subject: [PATCH 1/5] Fix build failure: include <glib.h>, not <glib/gtypes.h>
 directly.

---
 src/nm-openconnect-service.h |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/src/nm-openconnect-service.h b/src/nm-openconnect-service.h
index e454f82..684a5b3 100644
--- a/src/nm-openconnect-service.h
+++ b/src/nm-openconnect-service.h
@@ -25,8 +25,7 @@
 #ifndef NM_OPENCONNECT_PLUGIN_H
 #define NM_OPENCONNECT_PLUGIN_H
 
-#include <glib/gtypes.h>
-#include <glib-object.h>
+#include <glib.h>
 #include <nm-vpn-plugin.h>
 
 #define NM_TYPE_OPENCONNECT_PLUGIN            (nm_openconnect_plugin_get_type ())
-- 
1.7.7.2

