From 095b447c5a5935f2696d9db95e112435f9ee053f Mon Sep 17 00:00:00 2001
From: Christopher Aillon <caillon@redhat.com>
Date: Sat, 26 Mar 2011 17:24:52 -0700
Subject: [PATCH 2/2] replace nm_info with g_message

---
 src/nm-openconnect-service.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/nm-openconnect-service.c b/src/nm-openconnect-service.c
index 7ffb095..b2f8e83 100644
--- a/src/nm-openconnect-service.c
+++ b/src/nm-openconnect-service.c
@@ -356,7 +356,7 @@ nm_openconnect_start_openconnect_binary (NMOPENCONNECTPlugin *plugin,
 	}
 	g_ptr_array_free (openconnect_argv, TRUE);
 
-	nm_info ("openconnect started with pid %d", pid);
+	g_message ("openconnect started with pid %d", pid);
 
 	if (write(stdin_fd, props_cookie, strlen(props_cookie)) != strlen(props_cookie) ||
 		write(stdin_fd, "\n", 1) != 1) {
@@ -459,7 +459,7 @@ real_disconnect (NMVPNPlugin   *plugin,
 		else
 			kill (priv->pid, SIGKILL);
 
-		nm_info ("Terminated openconnect daemon with PID %d.", priv->pid);
+		g_message ("Terminated openconnect daemon with PID %d.", priv->pid);
 		priv->pid = 0;
 	}
 
-- 
1.7.4.1

