From c6d92c3e64898ea62789def04b86752dec904326 Mon Sep 17 00:00:00 2001
From: David Woodhouse <David.Woodhouse@intel.com>
Date: Thu, 10 Nov 2011 23:53:58 +0000
Subject: [PATCH 3/5] Drop g_thread_init()

---
 auth-dialog/main.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/auth-dialog/main.c b/auth-dialog/main.c
index 9892a3c..1a09c2f 100644
--- a/auth-dialog/main.c
+++ b/auth-dialog/main.c
@@ -1460,7 +1460,6 @@ int main (int argc, char **argv)
 		return 1;
 	}
 
-	g_thread_init (NULL);
 	gtk_init(0, NULL);
 
 	_ui_data = init_ui_data(vpn_name, options, secrets);
-- 
1.7.7.2

