From 02a75b1ce0ba38d9b447ce09dea61354cfdce9a0 Mon Sep 17 00:00:00 2001
From: Kjartan Maraas <kmaraas@gnome.org>
Date: Sun, 10 Jun 2012 19:09:37 +0200
Subject: [PATCH 15/20] Mark two strings for translation

---
 auth-dialog/main.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/auth-dialog/main.c b/auth-dialog/main.c
index 1dd9c39..f3eb493 100644
--- a/auth-dialog/main.c
+++ b/auth-dialog/main.c
@@ -667,9 +667,9 @@ static int nm_process_auth_form (void *cbdata, struct oc_auth_form *form)
 static char* get_title(const char *vpn_name)
 {
 	if (vpn_name)
-		return g_strdup_printf("Connect to VPN '%s'", vpn_name);
+		return g_strdup_printf(_("Connect to VPN '%s'"), vpn_name);
 	else
-		return g_strdup("Connect to VPN");
+		return g_strdup(_("Connect to VPN"));
 }
 
 typedef struct cert_data {
-- 
1.7.10.2

