commit 5f774ab660ecb6959a54b2f5c10dd7abbf77c9e4
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Wed Dec 17 12:35:47 2008 +0000

    Remove libgnomeui from configure.in. We don't ned it.

diff --git a/configure.in b/configure.in
index 66820c7..836ca37 100644
--- a/configure.in
+++ b/configure.in
@@ -69,10 +69,6 @@ if test x"$with_gnome" != xno; then
 	AC_SUBST(GLADE_CFLAGS)
 	AC_SUBST(GLADE_LIBS)
 
-	PKG_CHECK_MODULES(LIBGNOMEUI, libgnomeui-2.0)
-	AC_SUBST(LIBGNOMEUI_CFLAGS)
-	AC_SUBST(LIBGNOMEUI_LIBS)
-
 	PKG_CHECK_MODULES(GCONF, gconf-2.0)
 	AC_SUBST(GCONF_CFLAGS)
 	AC_SUBST(GCONF_LIBS)
