commit ba97bd5299e9a73e3d290591d7a6973040afce53
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Sat May 9 17:46:46 2009 +0100

    Don't abort on form entries being cached

diff --git a/src/nm-openconnect-service.c b/src/nm-openconnect-service.c
index 46c6fe1..4b09c8d 100644
--- a/src/nm-openconnect-service.c
+++ b/src/nm-openconnect-service.c
@@ -133,7 +133,7 @@ validate_one_property (const char *key, const char *value, gpointer user_data)
 	}
 
 	/* Did not find the property from valid_properties or the type did not match */
-	if (!info->table[i].name) {
+	if (!info->table[i].name && strncmp(key, "form:", 5)) {
 		g_set_error (info->error,
 		             NM_VPN_PLUGIN_ERROR,
 		             NM_VPN_PLUGIN_ERROR_BAD_ARGUMENTS,
