commit 6abbead21d7b4ca2c45294e41380690ed294028c
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Sat May 2 10:34:19 2009 +0100

    Add support for setting MTU in gconf

diff --git a/src/nm-openconnect-service.c b/src/nm-openconnect-service.c
index 1eae138..46c6fe1 100644
--- a/src/nm-openconnect-service.c
+++ b/src/nm-openconnect-service.c
@@ -36,7 +36,6 @@ static const char *openconnect_binary_paths[] =
 };
 
 #define NM_OPENCONNECT_HELPER_PATH		LIBEXECDIR"/nm-openconnect-service-openconnect-helper"
-#define NM_OPENCONNECT_UDP_ENCAPSULATION_PORT	0 /* random port */
 
 typedef struct {
 	const char *name;
@@ -55,6 +54,7 @@ static ValidProperty valid_properties[] = {
 	{ NM_OPENCONNECT_KEY_PRIVKEY,     G_TYPE_STRING, 0, 0 },
 	{ NM_OPENCONNECT_KEY_CERTSIGS,    G_TYPE_STRING, 0, 0 },
 	{ NM_OPENCONNECT_KEY_LASTHOST,    G_TYPE_STRING, 0, 0 },
+	{ NM_OPENCONNECT_KEY_MTU,         G_TYPE_STRING, 0, 0 },
 	{ NM_OPENCONNECT_KEY_AUTOCONNECT, G_TYPE_BOOLEAN, 0, 0 },
 	{ NULL,                           G_TYPE_NONE, 0, 0 }
 };
@@ -236,7 +236,7 @@ nm_openconnect_start_openconnect_binary (NMOPENCONNECTPlugin *plugin,
 	GPtrArray *openconnect_argv;
 	GSource *openconnect_watch;
 	gint	stdin_fd;
-	const char *props_vpn_gw, *props_cookie, *props_cacert;
+	const char *props_vpn_gw, *props_cookie, *props_cacert, *props_mtu;
 	
 	/* Find openconnect */
 	openconnect_binary = openconnect_binary_paths;
@@ -278,6 +278,7 @@ nm_openconnect_start_openconnect_binary (NMOPENCONNECTPlugin *plugin,
 	}
 
 	props_cacert = nm_setting_vpn_get_data_item (s_vpn, NM_OPENCONNECT_KEY_CACERT);
+	props_mtu = nm_setting_vpn_get_data_item (s_vpn, NM_OPENCONNECT_KEY_MTU);
 
 	openconnect_argv = g_ptr_array_new ();
 	g_ptr_array_add (openconnect_argv, (gpointer) (*openconnect_binary));
@@ -287,6 +288,11 @@ nm_openconnect_start_openconnect_binary (NMOPENCONNECTPlugin *plugin,
 		g_ptr_array_add (openconnect_argv, (gpointer) props_cacert);
 	}
 
+	if (props_mtu && strlen(props_mtu)) {
+		g_ptr_array_add (openconnect_argv, (gpointer) "--mtu");
+		g_ptr_array_add (openconnect_argv, (gpointer) props_mtu);
+	}
+
 	g_ptr_array_add (openconnect_argv, (gpointer) "--syslog");
 	g_ptr_array_add (openconnect_argv, (gpointer) "--cookie-on-stdin");
 
diff --git a/src/nm-openconnect-service.h b/src/nm-openconnect-service.h
index 71b74ba..67911b4 100644
--- a/src/nm-openconnect-service.h
+++ b/src/nm-openconnect-service.h
@@ -28,6 +28,7 @@
 #define NM_OPENCONNECT_KEY_XMLCONFIG "xmlconfig"
 #define NM_OPENCONNECT_KEY_CERTSIGS "certsigs"
 #define NM_OPENCONNECT_KEY_LASTHOST "lasthost"
+#define NM_OPENCONNECT_KEY_MTU "mtu"
 #define NM_OPENCONNECT_KEY_AUTOCONNECT "autoconnect"
 
 #define NM_OPENCONNECT_AUTHTYPE_CERT "cert"
